valhalla/userrequests/duration_utils.py

```diff
@@ -47,6 +49,42 @@ def get_request_duration_dict(request_dict):
     return req_durations
 
 
+def get_max_ipp_dict(request_dict, proposal_id):
+    proposal = Proposal.objects.get(pk=proposal_id)
+    request_durations = get_request_duration_sum(request_dict,
+                                                 proposal_id)
+    ipp_dict = {}
+    for tak, duration in request_durations.items():
+        time_allocation = proposal.timeallocation_set.get(semester=tak.semester,
+                                                           telescope_class=tak.telescope_class)
+        duration_hours = duration / 3600.0
+        ipp_available = time_allocation.ipp_time_available
+        max_ipp_allowable = min((ipp_available / duration_hours) + 1.0, MAX_IPP_LIMIT)
+        max_ipp_allowable = float("{:.3f}".format(max_ipp_allowable))
+        if tak.semester not in ipp_dict:
+            ipp_dict[tak.semester] = {}
+        ipp_dict[tak.semester][tak.telescope_class] = {'ipp_time_available': ipp_available,
+                                                       'ipp_limit': time_allocation.ipp_limit,
+                                                       'request_duration': duration_hours,
+                                                       'max_allowable_ipp_value': max_ipp_allowable,
+                                                       'min_allowable_ipp_value': MIN_IPP_LIMIT}
+    return ipp_dict
+
+
+def get_request_duration_sum(request_dict, proposal_id):
```

Comment:
This is super confusing and poorly named. request_dict is actually a list here, whereas it's a dictionary in every other function we have that takes request_dict